test/unit/mongoose_push_telemetry_metrics_test.exs

```diff
@@ -0,0 +1,124 @@
+defmodule MongoosePushTelemetryMetricsTest do
+  use ExUnit.Case, async: false
+  use Quixir
+  import Mock
```

Comment:
Let's not use `Mock` in new tests - we'll be trying to get rid of this library. Use `Mox` instead.

Reply:
I see a problem here - Mox can mock behaviours only and the behaviour we want to mock is `MongoosePush.Service`, but we never directly call any functions by `Service.push`. We choose a module based on the `:service` type and then call the required function from a chosen module. Additionally, we have more than one implementation of said behaviour (`APNS` and `FCM`). Making different behaviours for different services kills the purpose of a behaviour in a first place tough. Considering all that, I don't quite see a clean way to mock what we need.

Reply:
Then we simply need to move `:service` -> module mapping to configuration instead of having it hardcoded. That actually makes `:service` being more dynamic, which is also nice on it's own.

Reply:
Why would we make more then one behavior? I don't think I get you point, can you elaborate?

Reply:
Indeed, we have two implementations. Mock you be 3rd. 

We could change it so we have the following config file:

`prod.exs`:
```
config :mongoose_push, MongoosePush.Service:
  fcm: MongoosePush.Service.FCM,
  apns: MongoosePush.Service.APNS
```

`test.exs`:
```
config :mongoose_push, MongoosePush.Service:
  fcm: MongoosePush.Service.Mock,
  apns: MongoosePush.Service.Mock
```

Reply:
The config file suggestion made it clear for me, thanks

Reply:
Regarding that I decided to put
```
config :mongoose_push, MongoosePush.Service:
  fcm: MongoosePush.Service.FCM,
  apns: MongoosePush.Service.APNS
```
in `test.exs` as well, because it affect many tests and changing it should probably be a part of a story about replacing Mock with Mox